src/haskell/install.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+set -e
+
+BOOTSTRAP_HASKELL_GHC_VERSION="${VERSION:-"recommended "}"
+
+# Maybe install curl, gcc, make
+for x in curl gcc make; do
+    which $x > /dev/null || (apt update && apt install $x -y -qq)
+done
```

Comment:
It would be better to use a generic snippet to check other Features to do this.

Reply:
I believe @eitsupi is referring to using `check_packages` helper function instead of this logic.
https://github.com/devcontainers/features/blob/main/src/docker-in-docker/install.sh#LL86-L91C2

I think it would be nice to use `check_packages` in this Feature as well for consistency.
(Would also need to copy `apt_get_update` along with `check_packages`)

Reply:
Hrm, I'm not sure that those scripts do exactly what I want to do here. For example, they won't work correctly on a non-debian image where `gcc` is already installed.

Additionally, copying functions from one install.sh to another doesn't seem to be a great practice, is there a way to share common script snippets between features?

Reply:
> I'm not sure that those scripts do exactly what I want to do here. For example, they won't work correctly on a non-debian image where `gcc` is already installed.

Note that this repository is focused on debian-based containers at this time. Most Features will only work with debian-based images.

https://github.com/devcontainers/features/blob/1449f416a0d5c65fbd7e7bfac041eb43f1ecd98d/README.md?plain=1#L32

> Additionally, copying functions from one install.sh to another doesn't seem to be a great practice, is there a way to share common script snippets between features?

See devcontainers/spec#209